File: objetos/SistemaIrrigacao.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package objetos;

/**
 *
 * @author anacs
 */
public class SistemaIrrigacao {
    
    private boolean ligado;
    private double pressaoAgua;
    private String tipoIrrigacao;
    
    public SistemaIrrigacao(String tipoIrrigacao) {
        this.tipoIrrigacao = tipoIrrigacao;
        this.ligado = false;
        this.pressaoAgua = 0.0;
    }

    public void ligarIrrigacao() {
        ligado = true;
        pressaoAgua = 2.5;
        System.out.println("Sistema de irrigação " + tipoIrrigacao + " ligado.");
    }

    public void desligarIrrigacao() {
        ligado = false;
        pressaoAgua = 0.0;
        System.out.println("Sistema de irrigação " + tipoIrrigacao + " desligado.");
    }
    
    public boolean isLigado() {
        return ligado;
    }
    
    public double getPressaoAgua() {
        return pressaoAgua;
    }
    
    public String getTipoIrrigacao() {
        return tipoIrrigacao;
    }
    
}

File: python/sistema_irrigacao.py
class SistemaIrrigacao:
    def __init__(self, tipo_irrigacao):
        self.tipo_irrigacao = tipo_irrigacao
        self.ligado = False
        self.pressao_agua = 0.0

    def ligar_irrigacao(self):
        self.ligado = True
        self.pressao_agua = 2.5
        print(f"Sistema de irrigação {self.tipo_irrigacao} ligado.")

    def desligar_irrigacao(self):
        self.ligado = False
        self.pressao_agua = 0.0
        print(f"Sistema de irrigação {self.tipo_irrigacao} desligado.")

    def is_ligado(self):
        return self.ligado

    def get_pressao_agua(self):
        return self.pressao_agua

    def get_tipo_irrigacao(self):
        return self.tipo_irrigacao